✨ | [v3.1] Taxation Features, Advanced Logging & more features.

NamVr opened the pull request:
This PR is for v3.1.0 for NCE (Chat-Economy-Bot) to make improvements and add new features.

NamVr committed:
✨ | [Feat] Transfer Taxation

NamVr committed:
✨ | [Feat] Admin Currency Management

NamVr committed:
🐛 | [Fix] Typo error in currency command

NamVr committed:
⚠️ | [BREAKING] Search & Work Command Variables Updated

NamVr commented:
Logging is almost done (70%)! Hopeful release will be tomorrow :)

NamVr commented:
90% just implementation is left.

NamVr committed:
Till object comparer

NamVr commented:
My current work is pushed, I have still not figured out the part of comparing two databases together, that is the only remaining part!
I appreciate any contributions :)

StormPacer committed:
Reworked comparer + currency fix and example

StormPacer commented:
Object comparer works, fixed an issue with currency and logging setup should follow the example in currency.js

NamVr commented:
Tyvm, need to do some basic type corrections and just need to add timezone based streak command instead of +24 hour /daily.

nngggaming reviewed (dismissed):
### Grammatical Error
- File path: `interactions/slash/misc/currency.js`
- Lines: 145 to 147

Issue
- If `subCommand` is equal to `'remove'`, the following will be set as the embed description: 
> You have successfully removed [amount] [emoji] [name] to [user]

Fix
- The word 'to' should be changed to 'from':
> You have successfully removed [amount] [emoji] [name] from [user]
- Requested fix: https://pastebin.com/Dnhr7YiU (lines 145 to 147)

NamVr commented:
Thank you @nngggaming for suggestions, I will look into this ASAP.

NamVr committed:
Update node modules

NamVr committed:
Fixed word frequency for unscramble game

NamVr commented:
I will be removing logging features from v3.1 due to them not being completed, will finish these things later (v3.2)!

Feats for v3.1 will be:
- Taxes! Transfers can now have taxes!
- Admins can now generate currency with admin management commands (`/currency`)
- BREAKING change in search and work command variables!
- Fixed some typos/grammar.
- Updated node modules
- Fixed word frequency for unscramble the word game.

NamVr committed:
Currency management typo fix

NamVr committed:
Merge branch 'master' into dev
